napari/components/_viewer/view/buttons.py

```diff
@@ -48,8 +48,14 @@ def dragLeaveEvent(self, event):
         self.update()
 
     def dropEvent(self, event):
-        event.setDropAction(Qt.CopyAction)
         event.accept()
+        layer_name = event.mimeData().text()
+        layer = self.viewer.layers[layer_name]
+        if not layer.selected:
+            index = self.viewer.layers.index(layer)
+            self.viewer.layers.pop(index)
```

Comment:
you can just `self.viewer.layers.remove(layer)`

Reply:
fixed